src/display/api.js

```diff
@@ -3296,11 +3323,19 @@ class InternalRenderTask {
     }
     const { canvasContext, viewport, transform, background } = this.params;
 
+    let filterFactory = null;
+    if (isOffscreenCanvasSupported) {
+      filterFactory =
+        this._filterFactory ||
+        FilterFactory.getInstance({ ownerDocument: this._ownerDocument });
```

Comment:
Without doing that, a new FilterFactory is created on every rendering (it the same for the CanvasFactory).
In both case it's useless and I'd say that both should be a singleton. In the FilterFactory case, it leads to recreate a node in the document for each rendering and it makes the cache slightly useless.
In my first attempt, I almost did what you're proposing (i.e. follow the canvas example) but that led to this issue of having several svg nodes in the doc.

Reply:
> Without doing that, a new FilterFactory is created on every rendering (it the same for the CanvasFactory).
> In both case it's useless and I'd say that both should be a singleton.

I'm not sure if we can change the CanvasFactory at this point, without potentially introducing third-party breakage, and it'd be nice to have these things be *consistent*.

> In my first attempt, I almost did what you're proposing (i.e. follow the canvas example) but that led to this issue of having several svg nodes in the doc.

A related issue, as I've commented on elsewhere, is that there's never an attempt to do *any* clean-up of the document-wide data added by the FilterFactory. If multiple documents are opened that seems potentially problematic...

Reply:
> I'm not sure if we can change the CanvasFactory at this point, without potentially introducing third-party breakage, and it'd be nice to have these things be _consistent_.

On each FilterFactory creation, it'd then require to check if the svg node is in the document: I can do that but it's just useless and it'll induce an access to the DOM.
I agree with about the consistency but we should find a way to create those factories only one time.

Reply:
The other option, which I'm guessing would be overall simpler here, would be to move the `filterFactory` option to the `getDocument` level instead since it seems to me (unless I'm overlooking something) that it'd solve both problems here:
 - There would only be *one* FilterFactory-instance per PDF document opened.
 - We could clean-up the data added to the DOM, on both cleanup and destroy.